op-program/preimage/hints.go

```diff
@@ -0,0 +1,65 @@
+package preimage
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// HintWriter writes hints to an io.Writer (e.g. a special file descriptor, or a debug log),
+// for a pre-image oracle service to prepare specific pre-images.
+type HintWriter struct {
+	w io.Writer
+}
+
+var _ Hinter = (*HintWriter)(nil)
```

Comment:
Wouldn't the `HintWriter` and the `OracleClient` be in the client package so the client program can write out requests and the reader/server would be in the host module?

Reply:
Putting them in the same package makes testing between the client and server easier, since this is the inner-most layer where the client and server actually directly communicate data to each other.